src/Database/Database.php

```
@@ -573,10 +573,6 @@ public function createAttribute(string $collection, string $id, string $type, in
             }
         }
 
-        if($type === self::VAR_DATETIME && !\array_key_exists('datetime', $filters)) {
-            $filters[] = 'datetime';
-        }
-
```

any reason for this removal?

if so, we should add a test on what happens when an attribute is create that does not have the datetime filter. because it is required and under any circumstance should throw an error :+1:

---

So the `createAttribute` function has a parameter called `$filters` 

When we create a new attribute, it becomes the responsibility of the caller to pass in the relevant filters. I've updated the tests to ensure we pass in the  relevant filter when creating a datetime attribute

---

yeah, but is it possible to create that attribute without the filter?

---

In the end, you should be able to use the library and if no exception is thrown - means everything worked fine.

This will not be true in this case. So we either add the filter if its datetime or we throw an error if its missing.

---

Yes, it will allow you to create the attribute, however there will be an exception when you try to create a document since the filter is not applied.

---

I can think of 2 ways to make it more generic

1. We make `$filters` a required parameter instead of defaulting to `null`. This way, the caller must at least pass an empty array.
2. The second approach is to maintain a mapping of data types to their default filters
```php
$defaultFilters = [
   self::VAR_DATETIME => ['datetime'],
   self::VAR_STRING => [],
   .... 
];
```

And apply these filters in create attribute.. 


Even if we have to throw an exception or just use the default values, we need an array mapping like this to know which filters are required for which data types. 

LMK your thoughts...

---

1. will still make the attribute not work without passing it
2. can mess up adding additional filters where they order might matter

We should probably have a map of required filters and check. If missing we throw an exception 👍🏻